### PR TITLE
Add rocksdb

### PR DESCRIPTION
**Pull request by Yangqing**

### CODE REVIEW

**Comment by Yangqing** [comment]
Not sure what the protocol is, but seems to be common practice in previous PRs... ping @ocefpaf

**Comment by jakirkham** [comment]
So the normal protocol is to ping `@conda-forge/staged-recipes` for a review.

File: recipes/rocksdb/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+### Build
+export INSTALL_PATH=${PREFIX}
+make release
+make shared_lib
```
Comment (jakirkham) [inline recipes/rocksdb/build.sh]: Is there a `make check` or similar that we can run? Also would there be any value in building the static library as well?

File: recipes/rocksdb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{% set version = "5.8" %}
+
+package:
+  name: rocksdb
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Normally we do `{{ name }}-{{ version }}.tar.gz`. This is a bit nicer for people building the recipes locally as it allows `conda-build`'s source cache to keep track of `rocksdb` tarballs independently from those of other tarballs that may share the same version number.

File: recipes/rocksdb/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  url: https://github.com/facebook/rocksdb/archive/v{{ version }}.tar.gz
+  sha1: f0a5119bef4e1494e1af3871c288a1fe8200f304
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: This is fine, but we tend to use `sha256` if possible.

File: recipes/rocksdb/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - snappy
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Please add `toolchain` above.

Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Please change `snappy` to [`snappy 1.1.6|1.1.7`]( https://github.com/conda-forge/conda-forge.github.io/blob/b44a39dbcd919222dee6cae3d7ee82bdf44986ec/scripts/pin_the_slow_way.py#L85 ). This is the current way we are trying to pin `snappy` due to its [API/ABI compatibility]( https://abi-laboratory.pro/tracker/timeline/snappy/ ).

File: recipes/rocksdb/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  run:
+    - snappy
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Please change `snappy` to [`snappy 1.1.6|1.1.7`]( https://github.com/conda-forge/conda-forge.github.io/blob/b44a39dbcd919222dee6cae3d7ee82bdf44986ec/scripts/pin_the_slow_way.py#L85 ) here as well.

File: recipes/rocksdb/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/librocksdb.dylib            # [osx]
+    - test -f ${PREFIX}/lib/librocksdb.so               # [linux]
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Are there any headers we should check for as well? If we add static libraries, we should include them here too.

File: recipes/rocksdb/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+about:
+  home: https://github.com/facebook/rocksdb
+  license: Apache
```
Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Please include the version with license. Looks like this is Apache 2.0, correct? Also should the LevelDB license metadata be included as well?

Comment (jakirkham) [inline recipes/rocksdb/meta.yaml]: Please add `license_file: LICENSE.Apache` to package the Apache license file with the package.

Unfortunately we don't have a way currently to package multiple license files. So there isn't a way to include the LevelDB license file too yet. More details at issue ( https://github.com/conda/conda-build/issues/1247 ).

**Comment by jakirkham** [comment]
Thanks for the PR @Yangqing. RocksDB will be a nice addition to conda-forge. 😄

Added some comments in the PR above. Nothing to complicated hopefully. Please let me know if you have questions and/or when you are ready for another look.

**Pull request closed without merging**